### PR TITLE
[fix] Remove CompileTimeConstant from SafeIoException

### PR DESCRIPTION
**Pull request by ellisjoe**
## Before this PR
We have safe messages that are not compile time constants but part of our api. We can't upgrade to new versions of the library because we're unable to change the messages without breaking our api.

## After this PR
Back to the original behavior that allows all messages, and will allow us to upgrade.

### CODE REVIEW

**Comment by carterkozak** [comment]
Can you please provide an example? I have found this check incredibly valuable. For cases where we know data is safe, is it sufficient to include `SafeArg.of("message", safeNonConstantMessage)`?

**Review by carterkozak — CHANGES_REQUESTED** [review]
see PR comment

**Comment by iamdanfox** [comment]
More context:

- @pkoenig10's PR changed the exception message: https://github.com/palantir/conjure-java-runtime/pull/1014/files#diff-191491ccb0d50d8389983418d605cbbaL47
- Another library was parsing out the exception message and extracting this status code, so that PR broke behaviour

I'd like to spend a few minutes seeing if we can fix the parsing in that library before merging this though...

**Comment by carterkozak** [comment]
If the affected library is using `getMessage` and not `getLogMessage`, adding the old message as an additional parameter may unblock.

It is not good practice to parse exception messages, I would prefer not to optimize on that case.

**Comment by ellisjoe** [comment]
alternative proposal is here: https://github.com/palantir/conjure-java-runtime/pull/1023

**Comment by iamdanfox** [comment]
Not necessary as I've fixed up the internal product

**Pull request closed without merging**
